src/index.ts

```diff
@@ -275,8 +272,22 @@ export function sendDistributionMetric(name: string, value: number, ...tags: str
 
   if (currentMetricsListener !== undefined) {
     currentMetricsListener.sendDistributionMetric(name, value, false, ...tags);
-  } else {
-    logError("handler not initialized");
+    return;
+  }
+
+  _metricsQueue.push({ name, value, tags });
+}
+
+function sendQueueMetrics(listener: MetricsListener) {
+  while (_metricsQueue.length > 0) {
+    const metric = _metricsQueue.shift()!; // This will always exist.
+    const { name, value, metricTime, tags } = metric;
+    if (metricTime !== undefined) {
+      listener.sendDistributionMetricWithDate(name, value, metricTime, false, ...tags);
+      return;
```

Comment:
should this be `continue`?

Reply:
Yup, you are right, thanks – great catch!